lib/std/time.zig

```diff
@@ -59,6 +59,398 @@ pub fn sleep(nanoseconds: u64) void {
+        }
+
+        var week_day = @rem((3 + days), 7);
+        if (week_day < 0) {
+            week_day += 7;
+        }
+
+        var qc_cycles = @divTrunc(days, days_per_400y);
+        var rem_days = @as(i32, @intCast(@rem(days, days_per_400y)));
+        if (rem_days < 0) {
+            rem_days += days_per_400y;
+            qc_cycles -= 1;
+        }
+
+        var c_cycles = @divTrunc(rem_days, days_per_100y);
+        if (c_cycles == 4) {
+            c_cycles -= 1;
+        }
+        rem_days -= c_cycles * days_per_100y;
+
+        var q_cycles = @divTrunc(rem_days, days_per_4y);
+        if (q_cycles == 25) {
+            q_cycles -= 1;
+        }
+        rem_days -= q_cycles * days_per_4y;
+
+        var rem_years = @divTrunc(rem_days, 365);
+        if (rem_years == 4) {
+            rem_years -= 1;
+        }
+        rem_days -= rem_years * 365;
+
+        const leap: i32 = if (rem_years == 0 and (q_cycles != 0 or c_cycles == 0)) 1 else 0;
+        var year_day = rem_days + 31 + 28 + leap;
+        if (year_day >= 365 + leap) {
+            year_day -= 365 + leap;
+        }
+
+        var years = rem_years + 4 * q_cycles + 100 * c_cycles + 400 * qc_cycles;
+
+        var months: i32 = 0;
+        while (days_in_month[@as(usize, @intCast(months))] <= rem_days) : (months += 1) {
+            rem_days -= days_in_month[@as(usize, @intCast(months))];
+        }
+
+        if (months >= 10) {
+            months -= 12;
+            years += 1;
+        }
+
+        return .{
+            .year = @intCast(years + 2000),
+            .month = @enumFromInt(months + 3),
+            .day = @intCast(rem_days + 1),
+            .year_day = @intCast(year_day + 1),
+            .week_day = @enumFromInt(week_day),
+            .hour = @intCast(@divTrunc(rem_seconds, 3600)),
+            .minute = @intCast(@rem(@divTrunc(rem_seconds, 60), 60)),
+            .second = @intCast(@rem(rem_seconds, 60)),
+            .nanosecond = @intCast(@rem(nano_timestamp, ns_per_s)),
+            .nano_timestamp = nano_timestamp,
+            .tz_projection = tz_projection,
+        };
+    }
+
+    /// Compare two `DateTime`s.
+    pub fn order(self: DateTime, other: DateTime) std.math.Order {
+        return std.math.order(self.nano_timestamp, other.nano_timestamp);
+    }
+
+    pub const default_fmt = "%Y-%m-%dT%H%c%M%c%S%z";
+
+    /// %a Abbreviated weekday name (Sun)
+    /// %A Full weekday name (Sunday)
+    /// %b Abbreviated month name (Mar)
+    /// %B Full month name (March)
+    /// %c A colon ':'
+    /// %d Day of the month (01-31)
+    /// %H Hour in 24h format (00-23)
+    /// %I Hour in 12h format (01-12)
+    /// %j Day of the Year (001-366)
+    /// %m Month as a decimal number (01-12)
+    /// %M Minute (00-59)
+    /// %p AM or PM designation
+    /// %s Millisecond 891
+    /// %S Second (00-60)
+    /// %y Year, last two digits (00-99)
+    /// %Y Year
+    /// %z UTC offset in the form ±HHMM
+    /// %Z Time zone name.
```

Comment:
`%Z` typically gives you the abbreviation, not the (up-to-date) IANA db identifier. While this is nice for output, parsing the abbreviations is a pain, if not impossible since there are many ambiguous ones. I think it would be great to have a directive that gives the IANA identifier.